### PR TITLE
OpenAL: only enable on Windows + dynamically load DLL 

### PR DESCRIPTION
**Pull request by ligfx**
OpenAL using OpenAL Soft is generally worse than the other audio backends, and shouldn't be used or shown to users by default. The code's also a bit crufty, and could probably be cleaned up/sped up/made better if it had the leeway to depend on hardware extensions.

On Windows, this changes the backend to dynamically load `openal32.dll` rather than shipping with OpenAL Soft. This will pick up any system-installed library (like the Creative Labs OpenAL library, for instance) or a library in the same directory as Dolphin. If it can't be found, it won't be used or displayed to the end-user.

On non-Windows, this stops building the OpenAL backend.

### CODE REVIEW

**Comment by ligfx** [comment]
@shuffle2 @LAGonauta @MerryMage

**Comment by LAGonauta** [comment]
LGTM

----

A couple of days ago I discovered that virtual sinks on PulseAudio works better on Ubuntu 17.04, and Cubeb seems to send the full surround information to it (for system-wide HRTF) so there will be no loss of functionality.

Just compiled and tested your PR on desktop with a X-Fi, worked perfectly.

**Comment by ligfx** [comment]
NB that the behavior on Windows will be identical until the buildbot is cleaned, since `OpenAL32.dll` is still in the build directory.

@LAGonauta Can you check that it can pick up your system copy as well as the local one?

File: Source/Core/AudioCommon/OpenALStream.cpp [hunk 1]
```diff
@@ -2,70 +2,131 @@
...
+  if (!pContext)
   {
-    PanicAlertT("OpenAL: can't find sound devices");
+    palcCloseDevice(pDevice);
+    PanicAlertT("OpenAL: can't create context for device %s", defaultDeviceName);
+    return false;
   }
 
-  return bReturn;
+  // Used to determine an appropriate period size (2x period = total buffer size)
```
Comment (iwubcode) [inline Source/Core/AudioCommon/OpenALStream.cpp]: Did you intend to include these commented out lines?

**Comment by iwubcode** [comment]
I haven't tested this but the code looks good.  My only concern would be that we're including binary files in the git repository but considering these aren't new binary files, it is probably ok..

**Comment by ligfx** [comment]
@iwubcode It's actually removing binary files :)

**Comment by iwubcode** [comment]
Ah I missed that!  Unfortunately binary files are always there once added :(

That's not really a blocker to this PR, just a frustration with whoever added them in the first place.

**Comment by shuffle2** [comment]
lgtm except we should probably put in readme or somewhere, a link to a compatible/recommended version of the binary.

File: Source/Core/AudioCommon/OpenALStream.cpp (hunk 1 above)
Comment (ligfx) [inline Source/Core/AudioCommon/OpenALStream.cpp]: I honestly didn't think about it when I was reorganizing this method, but yeah these can be removed.

**Pull request merged**